src/main/java/uk/gov/gchq/palisade/service/request/DataRequestResponse.java

```diff
@@ -44,79 +43,81 @@ public DataRequestResponse() {
     }
 
     public DataRequestResponse token(final String token) {
-        requireNonNull(token, "The token cannot be null.");
-        this.token = token;
+        this.setToken(token);
         return this;
     }
 
-    public void setToken(final String token) {
-        token(token);
+    public DataRequestResponse resource(final LeafResource resource, final ConnectionDetail connectionDetail) {
+        this.setResource(resource, connectionDetail);
+        return this;
     }
 
-    public String getToken() {
-        requireNonNull(token, "The token has not been set.");
-        return token;
+    public DataRequestResponse resources(final Map<LeafResource, ConnectionDetail> resources) {
+        this.setResources(resources);
+        return this;
     }
 
-    public DataRequestResponse resource(final LeafResource resource, final ConnectionDetail connectionDetail) {
-        requireNonNull(resource, "The resource cannot be null.");
-        requireNonNull(connectionDetail, "The connection details cannot be null.");
-        if (null == resources) {
+    public void setResource(final LeafResource resource, final ConnectionDetail connectionDetail) {
```

Comment:
rename this method to addResource

Reply:
Done.  Note I had to change the resource mutator to make it work.